lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -252,6 +254,123 @@ void FIRRTLTypesLowering::visitDecl(InstanceOp op) {
   opsToRemove.push_back(op);
 }
 
+void FIRRTLTypesLowering::visitDecl(MemOp op) {
+
+  auto type = op.getDataTypeOrNull();
+  // Bail out if the memory is empty
```

Comment:
Should this also bail out if the data type is not an aggregate? I was under the impression that nothing would need to be done in that case (i.e. no call to create IR or `setBundleLowering`). If this simplification is correct, it might be nice to avoid doing unnecessary work when possible.

Reply:
Unfortunately, no. The problem is that while the memory type is a ground type, the ports will be aggregates after lowering and this method needs to setup some mapping for them in `setBundleLowering`. This will currently add the same dummy wires and set those as the lowering targets. However, this could be changed to not emit dummy wires for the memory ground type case.

Reply:
Got it. I guess I am just advocating for doing the simpler thing if possible in the memory ground type case.